### PR TITLE
update docs with Nix changes from status-react#9605

### PR DESCRIPTION
**Pull request by jakubgs**
Mostly renames `TARGET_OS` to `TARGET` and adds info about `make nix-gc` for space cleanup.
Also some small formatting fixes.

### CODE REVIEW

File: source/build_status/index.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ The `make shell TARGET_OS=<os>` script prepares and installs the following:
 
 *Note 2:* Specific tool versions used are maintained in the [.TOOLVERSIONS](https://github.com/status-im/status-react/blob/develop/.TOOLVERSIONS) file.
```
Comment (PombeirP) [inline source/build_status/index.md]: Didn't `.TOOLVERSIONS` go away?

**Comment by PombeirP** [comment]
Some paths also need updating, e.g. https://github.com/status-im/status.im/blob/nix-refactor-updates/source/build_status/desktop.md#L24

**Comment by PombeirP** [comment]
Unrelated, but please also delete this outdated section: https://github.com/status-im/status.im/blob/nix-refactor-updates/source/build_status/index.md#L176-L180

File: source/build_status/index.md (hunk 1 above)
Comment (jakubgs) [reply source/build_status/index.md]: Good catch!

**Comment by jakubgs** [comment]
I addressed your comments. Thanks!

**Pull request merged**